src/shipit_static_analysis/shipit_static_analysis/report/base.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import itertools
+from shipit_static_analysis.clang.tidy import ClangTidyIssue
+from shipit_static_analysis.clang.format import ClangFormatIssue
+
+COMMENT_FAILURE_SHORT = '''
+C/C++ static analysis found {defects_tidy} in this patch{extras_comments}.
+
+You can run this analysis locally with: `./mach static-analysis check path/to/file.cpp`
+'''
+COMMENT_FAILURE = '''
+C/C++ static analysis found {defects_total} in this patch{extras_comments}.
+ - {defects_tidy} found by clang-tidy
+ - {defects_format} found by clang-format
+
+You can run this analysis locally with: `./mach static-analysis check path/to/file.cpp` and `./mach clang-format -p path/to/file.cpp`
+'''
+BUG_REPORT = '''
+If you see a problem in this automated review, please report it here: http://bit.ly/2y9N9Vx
+'''
+COMMENT_DIFF_DOWNLOAD = '''
+
+A full diff for the formatting issues found by clang-format is provided here: {url}
```

Comment:
Formatting feedback is still enabled only on staging (without MozReview posting) and disabled in production, right?

Reply:
Yeap.
`clang-format` is always run, but we do not report its result on Mozreview yet. The results are always available in the admin emails though.